packages/contracts-bedrock/src/libraries/Arithmetic.sol

```diff
@@ -26,4 +26,32 @@ library Arithmetic {
     function cdexp(int256 _coefficient, int256 _denominator, int256 _exponent) internal pure returns (int256) {
         return (_coefficient * (FixedPointMathLib.powWad(1e18 - (1e18 / _denominator), _exponent * 1e18))) / 1e18;
     }
+
+    /// @notice Saturating addition.
+    /// @param _x The first value.
+    /// @param _y The second value.
+    /// @return z_ The sum of the two values, or the maximum value if the sum overflows.
+    /// @dev Returns `min(2 ** 256 - 1, x + y)`.
+    /// @dev Taken from Solady
+    /// https://github.com/Vectorized/solady/blob/63416d60c78aba70a12ca1b3c11125d1061caa12/src/utils/FixedPointMathLib.sol#L673
+    function saturatingAdd(uint256 _x, uint256 _y) internal pure returns (uint256 z_) {
+        /// @solidity memory-safe-assembly
```

Comment:
small nit but I prefer 
```
assembly ("memory-safe") {}
```

as its less like to get skipped over like with comments

Reply:
Done